src/sqlfluff/rules/L059.py

```diff
@@ -6,36 +6,118 @@
 
 from sqlfluff.core.parser.segments.raw import CodeSegment
 from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
-from sqlfluff.core.rules.doc_decorators import document_fix_compatible
-
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+)
 
+@document_configuration
 @document_fix_compatible
 class Rule_L059(BaseRule):
     """Unnecessary quoted identifier.
+    
+    This rule will fail if the quotes used to quote an identifier are unnecessary, 
+    or wrong considering the configuration.
+    
+    By default, the quotes are optional, but required to double-quotes when identifier 
+    is named as a keyword, or contains special characters.
 
     | **Anti-pattern**
     | In this example, a valid unquoted identifier,
     | that is also not a reserved keyword, is needlessly quoted.
 
     .. code-block:: sql
 
-        SELECT 123 as "foo"
+        SELECT 123 as "foo", 'sum'
 
     | **Best practice**
-    | Use unquoted identifiers where possible.
+    | If ``force_quote_identifier = False`` and ``preferred_quote_identifier = '"'`` (or unspecified, as these are the default)
+    | Use unquoted identifiers where possible, and use '"' whevener necessary.
```

Comment:
> whevener?

Yes, I'm very good at doing typos, unfortunately. `Whenever` is better.

I added the last line for cases when another quote identifier has been used instead of the preferred one.

For instance, with `force_quote_identifier = False` and `preferred_quote_identifier = '"'`, if you have an SQL query like:
```sql
SELECT 123 AS `SELECT`;
```
then, the L059 rule should fail, as the expected query should be:
```sql
SELECT 123 AS "SELECT";
```

Reply:
_Resolved as wanted behavior has changed_